### DISCUSSION TITLE
Add a way to show component while a component is loading

### DISCUSSION THREAD

**Discussion by omar2205** [withastro/roadmap#920] [Proposal]
# Summary

I have this component `<Quick client:only="react" />` it doesn't render anything until it hydrates. I want a way to show a spinner until it hydrates.

**Comment by florian-lefebvre** [comment]
I think you can use a fallback slot for this:
```astro
<ReactComponent client:only="react">
 <div slot="fallback">Loading...</div>
</ReactComponent>
```

**Reply by omar2205** [reply]
That worked nicely

**Reply by florian-lefebvre** [reply]
Glad it worked, feel free to mark my answer as the solution!

**Reply by carvinlo** [reply]
Duplicate of https://github.com/withastro/roadmap/discussions/416#discussioncomment-9263363

**Reply by florian-lefebvre** [reply]
FYI addition to docs is being tracked at https://github.com/withastro/docs/issues/8805
